File: Samples/JotFormTest/Classes/Controller/GetAppKeyViewController.m
//
//  GetAppKeyViewController.m
//  JotFormTest
//
//  Created by Interlogy, LLC on 1/5/14.
//  Copyright (c) 2014 Interlogy, LLC. All rights reserved.
//

#import "GetAppKeyViewController.h"
#import "SampleListViewController.h"
#import "SharedData.h"
#import "SVProgressHUD.h"
#import <JotForm_iOS/JotForm.h>
#import "Common.h"

@interface GetAppKeyViewController ()

@property (nonatomic,weak) IBOutlet UITextField *usernameTextField;
@property (nonatomic,weak) IBOutlet UITextField *passwordTextField;

@end

@implementation GetAppKeyViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    self.title = @"Get App Key";
    [self showAlertView];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - user definition method

- (void) showSampleListViewController
{
    SampleListViewController *sampleListVc = [[SampleListViewController alloc] initWithNibName:@"SampleListViewController" bundle:nil];
    
    [self.navigationController pushViewController:sampleListVc animated:YES];
}

- (void) showAlertView
{
    
    UIAlertController *alertView = [UIAlertController
                                    alertControllerWithTitle:@"JotFormAPISample"
                                    message:@"Do you have your Jotform account?"
                                    preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *noButton = [UIAlertAction
                                   actionWithTitle:@"No, I have an API key"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction *action){
                                       if ([API_KEY isEqualToString:@""]) {
                                           
                                           UIAlertController *alertViewCancel = [UIAlertController
                                                                           alertControllerWithTitle:@"JotFormAPISample"
                                                                           message:@"Please put your API key in Common.h 12 line." preferredStyle:UIAlertControllerStyleAlert];
                                           
                                           UIAlertAction *noCancelButton = [UIAlertAction
                                                                      actionWithTitle:@"No, I have an API key"
                                                                      style:UIAlertActionStyleDefault
                                                                      handler:^(UIAlertAction *action){
                                                                      }];
                                           [alertViewCancel addAction:noCancelButton];
                                           [self presentViewController:alertViewCancel animated:YES completion:nil];
                                       } else {
                                           
                                           SharedData *sharedData = [SharedData sharedData];
                                           [sharedData initAPIClient:API_KEY];
                                           
                                           [self showSampleListViewController];
                                       }

                                   }];
    
    UIAlertAction *yesButton = [UIAlertAction
                               actionWithTitle:@"Yes, I have an API key"
                               style:UIAlertActionStyleDefault
                               handler:^(UIAlertAction *action){
                                   exit(0);
                               }];
    
    [alertView addAction:yesButton];
    [alertView addAction:noButton];
    [self presentViewController:alertView animated:YES completion:nil];
}

#pragma mark - IBAction

- (IBAction) getAppKeyButtonClicked : (id) sender
{
    JotForm *apiClient = [[JotForm alloc] init];
    
    NSString *username = self.usernameTextField.text;
    
    if ( [username isEqualToString:@""] )
        [self.usernameTextField becomeFirstResponder];
    
    NSString *password = self.passwordTextField.text;
    
    if ( [password isEqualToString:@""] )
        [self.passwordTextField becomeFirstResponder];
    
    [self.usernameTextField resignFirstResponder];
    [self.passwordTextField resignFirstResponder];
    
    [SVProgressHUD showWithStatus:@"Getting app key..."];
    
    NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] init];
    [userInfo setObject:username forKey:@"username"];
    [userInfo setObject:password forKey:@"password"];
    [userInfo setObject:@"JotFormAPISample" forKey:@"appName"];
    [userInfo setObject:@"full" forKey:@"access"];

    [apiClient login:userInfo onSuccess:^(id result) {
        if ( result != nil ) {
            NSInteger responseCode = [[result objectForKey:@"responseCode"] integerValue];
            
            if ( responseCode == 200 || responseCode == 206 ) {
                
                id content = [result objectForKey:@"content"];
                
                SharedData *sharedData = [SharedData sharedData];
                [sharedData initAPIClient:[content objectForKey:@"appKey"]];
                
                [self showSampleListViewController];
            }
        }
        
        [SVProgressHUD dismiss];
    } onFailure:^(id error) {
          [SVProgressHUD dismiss];
    }];
}

#pragma mark - Jotform delegate

- (void) loginFinish : (id) result
{
    if ( result != nil ) {
        NSInteger responseCode = [[result objectForKey:@"responseCode"] integerValue];
        
        if ( responseCode == 200 || responseCode == 206 ) {
            id content = [result objectForKey:@"content"];
            
            SharedData *sharedData = [SharedData sharedData];
            [sharedData initAPIClient:[content objectForKey:@"appKey"]];
            
            [self showSampleListViewController];
        }
    }
    
    [SVProgressHUD dismiss];
}

- (void) loginFail : (id) error
{
    [SVProgressHUD dismiss];
}



@end

File: Samples/JotFormTest/Classes/Controller/GetAppKeyViewController.swift
//
//  GetAppKeyViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 1/31/18.
//  Copyright © 2018 wang. All rights reserved.
//

import UIKit
import Foundation
import JotForm_iOS


class GetAppKeyViewController: UIViewController {
    var apiClient: JotForm?
    
    @IBOutlet weak var usernameTextField: UITextField?
    @IBOutlet weak var passwordTextField: UITextField?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        apiClient = JotForm(apiKey: "", debugMode: false, euApi: false)
      
        title = "Get App Key"
        showAlertView()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(false)
        usernameTextField?.text = ""
        passwordTextField?.text = ""
    }
    
    func showSampleListViewController() {
       /* let sampleListVc = SampleListViewController(nibName: "SampleListViewController", bundle: nil)
        navigationController?.pushViewController(sampleListVc as? UIViewController ?? UIViewController(), animated: true) */
    }
    
    func showAlertView() {
        let alertView = UIAlertController(title: "JotFormAPISample", message: "Do you have an API key?", preferredStyle: .alert)
        
        let yesButton = UIAlertAction(title: "Yes", style: .default, handler: {(_ action: UIAlertAction) -> Void in
            if (API_KEY == "") {
                let alertViewCancel = UIAlertController(title: "JotFormAPISample", message: "Please put your API key in Common.h.", preferredStyle: .alert)
                
                let cancelButton = UIAlertAction(title: "Ok", style: .default, handler: {(_ action: UIAlertAction) -> Void in
                })
                
                alertViewCancel.addAction(cancelButton)
                self.present(alertViewCancel, animated: true, completion: nil)
            } else {
                SharedData().initAPIClient(API_KEY, euApi: EU_API)
                self.showSampleListViewController()
            }
        })
        
        let noButton = UIAlertAction(title: "No", style: .default, handler: {(_ action: UIAlertAction) -> Void in
        })
        
        alertView.addAction(yesButton)
        alertView.addAction(noButton)
        self.present(alertView, animated: true, completion: nil)
    }
    
    @IBAction func getAppKeyButtonClicked(_ sender: Any) {
        let storage = HTTPCookieStorage.shared
        for cookie: HTTPCookie in (storage.cookies)! {
            storage.deleteCookie(cookie)
        }
        
        usernameTextField?.resignFirstResponder()
        passwordTextField?.resignFirstResponder()
        
      //  SVProgressHUD.show(withStatus: "Getting app key...")
        var userInfo = [String: Any]()
        userInfo["username"] = usernameTextField?.text
        userInfo["password"] = passwordTextField?.text
        userInfo["appName"] = "JotFormAPISample"
        userInfo["access"] = "full"

        apiClient?.login(userInfo, onSuccess: {(_ result: AnyObject) -> Void in
            let responseCode: Int = result["responseCode"]  as! Int
            if responseCode == 200 || responseCode == 206 {
                let content = result["content"] as AnyObject
                let appKey = content["appKey"] as! String
                self.checkEuServer(appKey)
            }
        }, onFailure: {(_ error: Any) -> Void in
            // SVProgressHUD.dismiss()
        })
    }
    
    func checkEuServer(_ appKey: String) {
        apiClient?.checkEUserver(appKey, onSuccess: {(_ result: AnyObject) -> Void in
           // var isEuServer: Bool = result["content"]["euOnly"] != 0
            //  [[SharedData sharedData] initAPIClient:appKey euApi:isEuServer];
           // self.showSampleListViewController()
           // SVProgressHUD.dismiss()
        }, onFailure: {(_ error: Error?) -> Void in
           // SVProgressHUD.dismiss()
        })
    }
}
